Analysis/Tasks/PWGHF/HFMCValidation.cxx

```diff
@@ -0,0 +1,202 @@
+     {"hCouterPerCollisionDplus", "Event counter - DPlus; Events Per Collision; entries", {HistType::kTH1F, {{10, -0.5, +9.5}}}},
+     {"hCouterPerCollisionDstar", "Event counter - Dstar; Events Per Collision; entries", {HistType::kTH1F, {{10, -0.5, +9.5}}}},
+     {"hCouterPerCollisionLambdaC", "Event counter - LambdaC; Events Per Collision; entries", {HistType::kTH1F, {{10, -0.5, +9.5}}}}}};
+
+  void process(aod::McCollision const& mccollision, aod::McParticles const& particlesMC)
+  {
+    int cPerCollision = 0;
+    int cBarPerCollision = 0;
+    int bPerCollision = 0;
+    int bBarPerCollision = 0;
+    double sumPxDau, sumPyDau, sumPzDau;
+    bool momentumCheck;
+    double pxDiff, pyDiff, pzDiff;
+
+    //Particles and their decay checked in the second part of the task
+    std::array<int, 4> PDGArrayParticle = {411, 413, 421, 4122};
+    std::array<std::array<int, 3>, 4> arrPDGFinal = {{{211, 211, -321}, {211, 211, -321}, {-321, 211, 0}, {2212, -321, 211}}};
+    int counter[4] = {0, 0, 0, 0};
+    std::vector<int> listDaughters;
+
+    for (auto& particle : particlesMC) {
+      int particlePdgCode = particle.pdgCode();
+      if (particle.mother0() < 0) {
+        continue;
+      }
+      auto mother = particlesMC.iteratorAt(particle.mother0());
+      if (particlePdgCode != mother.pdgCode()) {
+        switch (particlePdgCode) {
+          case 4:
+            cPerCollision++;
+            break;
+          case -4:
+            cBarPerCollision++;
+            break;
+          case 5:
+            bPerCollision++;
+            break;
+          case -5:
+            bBarPerCollision++;
+            break;
+        }
+      }
+
+      sumPxDau = 0;
+      sumPyDau = 0;
+      sumPzDau = 0;
+      momentumCheck = 1;
+      listDaughters.clear();
+
+      // Checking the decay of the particles and the momentum conservation
+      for (int iD = 0; iD < PDGArrayParticle.size(); iD++) {
+        if (std::abs(particlePdgCode) == PDGArrayParticle[iD]) {
+          RecoDecay::getDaughters(particlesMC, particle.globalIndex(), &listDaughters, arrPDGFinal[iD], -1);
+          int arrayPDGsize = arrPDGFinal[iD].size() - std::count(arrPDGFinal[iD].begin(), arrPDGFinal[iD].end(), 0);
+          if (listDaughters.size() == arrayPDGsize) {
+            counter[iD]++;
+          }
+          for (int i = 0; i < listDaughters.size(); i++) {
+            auto daughter = particlesMC.iteratorAt(listDaughters.at(i));
+            sumPxDau += daughter.px();
+            sumPyDau += daughter.py();
+            sumPzDau += daughter.pz();
+          }
+          pxDiff = particle.px() - sumPxDau;
+          pyDiff = particle.py() - sumPyDau;
+          pzDiff = particle.pz() - sumPzDau;
+          if (std::abs(pxDiff) > 0.001 || std::abs(pyDiff) > 0.001 || std::abs(pzDiff) > 0.001) {
+            momentumCheck = 0;
+          }
+          double pDiff = RecoDecay::P(pxDiff, pyDiff, pzDiff);
+          double ptDiff = RecoDecay::Pt(pxDiff, pyDiff);
```

Comment:
@ginnocen The PR looks ready to me.
The only doubt I have is whether we want to plot the magnitude of the vectorial difference (as done now) or just the scalar difference of the two momenta magnitudes (which is smaller or equal).

Reply:
hi @vkucera I think the vectorial difference is ok